Core/include/Acts/Seeding/SeedFinder.ipp

```diff
@@ -247,7 +251,7 @@ SeedFinder<external_spacepoint_t, platform_t>::getCompatibleDoublets(
     // the iterator so we don't need to look at the other SPs again
     for (; min_itr != otherSPs.end(); ++min_itr) {
       const auto& otherSP = *min_itr;
-      if constexpr (candidateType == Acts::SpacePointCandidateType::eBottom) {
+      if (isBottomCandidate) {
```

Comment:
for my understanding, @paulgessinger when is this evaluated? should we keep the `if constexpr`?